drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -3828,8 +3828,8 @@ bool RenderingDeviceDriverVulkan::pipeline_cache_create(const Vector<uint8_t> &p
 
 	// Parse.
 	{
-		if (p_data.size() <= (int)sizeof(PipelineCacheHeader)) {
-			WARN_PRINT("Invalid/corrupt pipelines cache.");
+		if (p_data.size() != (int)sizeof(PipelineCacheHeader)) {
```

Comment:
I can change it back to `<=` if it was indeed intentional to treat a bigger buffer as supported and attempt parsing it.

Reply:
That sounds like it'd make it fail any time the data is not exactly the size of the header. The condition is for ensuring it has enough bytes to parse at least the header.

Reply:
At first I only singled out the `size() == 0` case, so that it doesn't warn when there's no data (which is the case currently, it just loads the file and sends the data to the function without further check).

But then I still got warnings after adding a 3D node to my project, presumably because the data was now bigger than the previous pipelines cache generated from an empty scene? Might have been with my `!=` change though, so I'll test again with `<=`.

Reply:
I can say in isolation the code doing the <= check is the intended approach at least. I believe there's other reasons why it keeps complaining it's corrupted, but this condition can't be it: it's just checking if there's enough bytes to parse the header.

Reply:
I reverted that change and now only handle the `size() == 0` case specifically.

That solves the warning when editing a project without a pre-existing cache, and doesn't seem to trigger warnings when modifying the project (from a quick test). So the warnings I saw earlier were likely from my changing the check to `!=`.